registry/storage/garbagecollect.go

```diff
@@ -47,6 +54,32 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 		}
 
 		err = manifestEnumerator.Enumerate(ctx, func(dgst digest.Digest) error {
+			if opts.RemoveUntagged {
+				tags, err := repository.Tags(ctx).Lookup(ctx, distribution.Descriptor{Digest: dgst})
+				if err != nil {
+					return fmt.Errorf("failed to retrieve tags for digest %v: %v", dgst, err)
+				}
+				if len(tags) == 0 {
+					// check does the path exist
```

Comment:
should I delete all these rows
```
					// check does the path exist
					_, err := pathFor(manifestRevisionPathSpec{name: repoName, revision: dgst})
					if err != nil {
						return err
					}
```

because this is not really doing anything else than making string for path, it does not even check does it exist and so on? What you think